build_test.go

```diff
@@ -180,12 +175,91 @@ launch = true
 				},
 			}))
 
+			Expect(calculator.SumCall.Receives.Paths).To(Equal([]string{
+				filepath.Join(workingDir, "app", "assets"),
+			}))
+
 			Expect(buildProcess.ExecuteCall.CallCount).To(Equal(0))
 
 			Expect(buffer.String()).To(ContainSubstring("Some Buildpack some-version"))
 			Expect(buffer.String()).To(ContainSubstring("Reusing cached layer"))
 		})
 
+		context("when there is a lib/assets directory", func() {
+			it.Before(func() {
+				Expect(os.RemoveAll(filepath.Join(workingDir, "app", "assets"))).To(Succeed())
+				Expect(os.MkdirAll(filepath.Join(workingDir, "lib", "assets"), os.ModePerm)).To(Succeed())
+			})
+
+			it("uses that directory to calculate the checksum", func() {
+				_, err := build(packit.BuildContext{
+					WorkingDir: workingDir,
+					CNBPath:    cnbDir,
+					Stack:      "some-stack",
+					BuildpackInfo: packit.BuildpackInfo{
+						Name:    "Some Buildpack",
+						Version: "some-version",
+					},
+					Layers: packit.Layers{Path: layersDir},
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(calculator.SumCall.Receives.Paths).To(Equal([]string{
+					filepath.Join(workingDir, "lib", "assets"),
+				}))
+			})
+		})
+
+		context("when there is a vendor/assets directory", func() {
+			it.Before(func() {
+				Expect(os.RemoveAll(filepath.Join(workingDir, "app", "assets"))).To(Succeed())
+				Expect(os.MkdirAll(filepath.Join(workingDir, "vendor", "assets"), os.ModePerm)).To(Succeed())
+			})
+
+			it("uses that directory to calculate the checksum", func() {
+				_, err := build(packit.BuildContext{
+					WorkingDir: workingDir,
+					CNBPath:    cnbDir,
+					Stack:      "some-stack",
+					BuildpackInfo: packit.BuildpackInfo{
+						Name:    "Some Buildpack",
+						Version: "some-version",
+					},
+					Layers: packit.Layers{Path: layersDir},
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(calculator.SumCall.Receives.Paths).To(Equal([]string{
+					filepath.Join(workingDir, "vendor", "assets"),
+				}))
+			})
+		})
+
+		context("when there is a vendor/assets directory", func() {
```

Comment:
should probably read `"when there is a app/assets directory"`, right?

Reply:
This test is about missing app/assets and present vendor/assets, and it expects to be successful.

Reply:
But below there is the line `Expect(os.RemoveAll(filepath.Join(workingDir, "app", "assets"))).To(Succeed())` which does not refer to `vendor/assets` but `app/assets`.

Reply:
Aaa... I see now. There are 3 similar tests for 3 folders, and 2nd and 3rd have the same text. Github shows me lines 235-238, where all is OK, but in next test not OK. It came from `paketo/rails-assets` as is and I missed it. I fixed.